### PR TITLE
Update erc20 migration branch

### LINKED ISSUE #546: migrate erc20 contracts
[reference ERC20 implementation](https://github.com/starkware-libs/cairo/blob/main/crates/cairo-lang-starknet/test_data/erc20.cairo)

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes #546. Updates #581 with the current cairo syntax.

#### PR Checklist

- [x] Tests
- [ ] Tried the feature on a public network
- [ ] Documentation

### CODE REVIEW

**Review by Amxx — APPROVED** [review]
LGTM

File: src/openzeppelin/tests/test_erc20.cairo [hunk 1]
```diff
@@ -139,33 +164,33 @@ fn test__transfer() {
 
 #[test]
 #[available_gas(2000000)]
-#[should_panic]
+#[should_panic(expected = ('u256_sub Overflow', ))]
```
Comment (martriay) [inline src/openzeppelin/tests/test_erc20.cairo:167]: why is `Overflow` uppercase?

File: src/openzeppelin/tests/test_erc20.cairo [hunk 2]
```diff
@@ -201,10 +226,9 @@ fn test_transfer_from_doesnt_consume_infinite_allowance() {
 
     let recipient: ContractAddress = contract_address_const::<2>();
     let spender: ContractAddress = contract_address_const::<3>();
-    let amount: u256 = u256_from_felt(100);
+    let amount: u256 = u256_from_felt252(100);
     let max_u256: u256 = u256 {
```
Comment (martriay) [inline src/openzeppelin/tests/test_erc20.cairo:230]: should we add this as a constant/function until Hadrien's PR is merged upstream?

**Review by martriay — APPROVED** [review]
looks good! shouldn't this PR target `cairo-1`?

File: src/openzeppelin/tests/test_erc20.cairo (hunk 1 above)
Comment (andrew-fleming) [reply src/openzeppelin/tests/test_erc20.cairo:167]: It's the returned error from the Cairo lib: https://github.com/starkware-libs/cairo/blob/main/corelib/src/integer.cairo#L889

**Comment by andrew-fleming** [comment]
> shouldn't this PR target cairo-1?

Yes, good catch! I was originally trying to merge it to the branch of the original PR. It's fixed

**Pull request merged**